CAP/gap/MethodRecord.gi

```diff
@@ -79,15 +215,49 @@ ProjectiveLift := rec(
   filter_list := [ "morphism", "morphism" ],
   io_type := [ [ "alpha", "beta" ], [ "alpha_source", "beta_source" ] ],
   cache_name := "ProjectiveLift",
-  return_type := "morphism_or_fail",
+  pre_function := function( iota, tau )
+    local value;
+    
+    value := IsEqualForObjects( Range( iota ), Range( tau ) );
+    
+    if value = fail then
+        
+        return [ false, "cannot decide whether the two morphisms have equal ranges" ];
+        
+    elif value = false then
+        
+        return [ false, "the two morphisms must have equal ranges" ];
+        
+    fi;
+    
+    return [ true ];
+  end,
+  return_type := "morphism",
```

Comment:
This seems wrong, as the documentation says this can return fail.

Reply:
Which part of the documentation are you referring to exactly? In my version, neither `ProjectiveLift` nor `AddProjectiveLift` mentions `fail`.

Reply:
And if it did, it would simply be wrong.

Reply:
Sorry, I was probably looking at the docs for Lift/Colift.